src/dataflow/shared/ingestion.py

```
@@ -0,0 +1,543 @@
+        Path to the saved file
+
+    Raises:
+        DataIngestionError: If saving to file fails
+    """
+    try:
+        # Ensure directory exists
+        os.makedirs(os.path.dirname(os.path.abspath(file_path)), exist_ok=True)
+
+        is_binary = "b" in mode
+
+        if isinstance(data, (dict, list)):
```

Ruff (0.8.2)

Use `X | Y` in `isinstance` call instead of `(X, Y)`

Convert to `X | Y`

(UP038)